promql/parser/ast.go

```diff
@@ -333,10 +333,13 @@ func Walk(v Visitor, node Node, path []Node) error {
 	if v, err = v.Visit(node, path); v == nil || err != nil {
 		return err
 	}
-	path = append(path, node)
+	var pathToHere []Node // Initialized only when needed.
```

Comment:
not sure if it'd be more efficient with a bool and `path = append(path, node) only when false`

Reply:
Minimally, I think. I expect bool would consume at least 4 bytes on the stack, maybe 8 after alignment.  I think having a separate pointer is clearer, so `path` only has one meaning.